tests/utils_/test_parse_uri.py

```diff
@@ -0,0 +1,90 @@
+from unittest.mock import MagicMock, patch
+
+import pymongo
+from django.test import SimpleTestCase
+
+from django_mongodb import parse_uri
+
+URI = "mongodb+srv://myDatabaseUser:D1fficultP%40ssw0rd@cluster0.example.mongodb.net/myDatabase?retryWrites=true&w=majority&tls=false"
+
+
+class ParseURITests(SimpleTestCase):
+    def setUp(self):
+        self.srv_record = MagicMock()
+        self.srv_record.target.to_text.return_value = "cluster0.example.mongodb.net"
+        self.patcher = patch("dns.resolver.resolve", return_value=[self.srv_record])
+        self.mock_resolver = self.patcher.start()
+        self.addCleanup(self.patcher.stop)
```

Comment:
Is the mocking needed? If I comment out the method (even with networking disabled) or change the return value to e.g. cluster1, I see no failures. Also, the [DNS resolving code](https://github.com/mongodb/mongo-python-driver/blob/1b89da4829945359ae99ecfb579104d2d28f7b27/pymongo/uri_parser.py#L557-L559) only seems applicable to srv URLs, but some tests with non-srv URIs are decorated with `@patch("dns.resolver.resolve")`. If any of this is needed, please add some explanatory comments.

Reply:
Yes, but only on one of the tests, e526ab995b5b6a0b1ed61b7cca05d39ebc2906b1

Reply:
Please add some explanatory comments about the reason mocking is needed.

Reply:
Done in 6ec2d516255d59ae408daa4d20ea978c8b73ec11

Reply:
Ok, now I can reproduce the crash by removing  `@patch`. But that seems sufficient... as far as I can tell, `setUp()` is useless. I think it would be slightly more clear if the comment/patch were inside the test method:
```
        with patch("dns.resolver.resolve"):
            settings_dict = parse_uri(URI)
```

Reply:
Well played, 1d7e4456af5a8271ccdc6c02b40d3c294619895d, 3549729392a2c0e88e6699e0516980a4b69e332c